optimuspy.py

```diff
@@ -134,16 +152,8 @@ def main(instance_name: str, cube_name: str, view_name: str, process_name: str,
     with TM1Service(**tm1_args) as tm1:
         original_performance_monitor_state = retrieve_performance_monitor_state(tm1)
         activate_performance_monitor(tm1)
-        all_tm1_cubes = filter(lambda c: not c.startswith("}"), tm1.cubes.get_all_names())
-
-        if cube_name is not None:
-            if cube_name in all_tm1_cubes:
-                model_cubes = [cube_name]
-            else:
-                logging.error(f"Provided cube '{cube_name}' does not exist")
-                exit(1)
-        else:
-            model_cubes = all_tm1_cubes
+        
+        model_cubes = get_cubes_to_optimize(tm1, cube_name)
 
         for cube_name in model_cubes:
```

Comment:
the loop would exit and produce the following output:

    Provided cube 'I dont exist' does not exist, nothing to optimize
    Finished successfully
    
    Process finished with exit code 0